js/src/main/scala/scalajson/ast/JValue.scala

```diff
@@ -77,22 +77,26 @@ object JNumber {
   def apply(value: Float): JValue = value match {
     case n if java.lang.Float.isNaN(n) => JNull
     case n if n.isInfinity => JNull
-    case _ => JNumber(value.toString)
+    case _ => new JNumber(value.toString)
   }
+
+  def fromString(value: String): Option[JNumber] =
```

Comment:
I am thinking of leaving in the `fromString` method for all versions and aliasing it to `apply` for Scala 2.10.x and Scala 2.11.x, that way for a codebase that needs to targets all Scala versions you can use `fromString`